src/main/java/ch/njol/skript/lang/Section.java

```diff
@@ -108,6 +108,26 @@ protected void loadCode(SectionNode sectionNode) {
 	 */
 	@SafeVarargs
 	protected final Trigger loadCode(SectionNode sectionNode, String name, Class<? extends Event>... events) {
+		return loadCode(sectionNode, name, null, events);
+	}
+
+	/**
+	 * Loads the code in the given {@link SectionNode},
+	 * appropriately modifying {@link ParserInstance#getCurrentSections()}.
+	 *
+	 * This method differs from {@link #loadCode(SectionNode)} in that it
+	 * is meant for code that will be executed in a different event.
+	 *
+	 * @param sectionNode The section node to load.
+	 * @param name The name of the event(s) being used.
+	 * @param afterLoading A Runnable to execute after the SectionNode has been loaded.
+	 * This occurs before {@link ParserInstance} states are reset.
+	 * @param events The event(s) during the section's execution.
+	 * @return A trigger containing the loaded section. This should be stored and used
+	 * to run the section one or more times.
+	 */
+	@SafeVarargs
+	protected final Trigger loadCode(SectionNode sectionNode, String name, @Nullable Runnable afterLoading, Class<? extends Event>... events) {
```

Comment:
The IDE will give an error about an ambiguous method call. I don't think any users will actually be doing `loadCode(section, name, null, PlayerEggThrowEvent.class)` as they can just do `loadCode(section, name, PlayerEggThrowEvent.class)`.

I don't really like the idea of switching around the parameters. If you really want, we could split it into two methods:
```java
protected final loadCode(SectionNode sectionNode, String name, Runnable afterLoading, Class<? extends Event>... events)
private final loadCode_i(SectionNode sectionNode, String name, @Nullable Runnable afterLoading, Class<? extends Event>... events)
```
Essentially, only the internal method would allow a nullable parameter.

Reply:
I mean with the internal method, null would only be passed internally for the internal method with afterLoading marked as nullable. In that case, the user should *never* pass null (and if they do, that's their problem). Really, the only reason the method is even marked as nullable is so that the code doesn't need to be duplicated for the method calls (which is why I proposed an internal method if you think we should be worried enough about it)

Reply:
As long as `loadCode(SectionNode, String, Runnable, PlayerEggThrowEvent.class)` won't be ambiguous to `loadCode(SectionNode, String, Runnable, PlayerEggThrowEvent.class, PlayerJoinEvent.class)` and 

`loadCode(SectionNode, String, PlayerEggThrowEvent.class, PlayerJoinEvent.class)` to `loadCode(SectionNode, String, PlayerEggThrowEvent.class, PlayerJoinEvent.class)` then it should be alright.